core/node.go

```diff
@@ -28,19 +33,37 @@ const (
 	maxReceiveBatch = 100
 	// maxShareBatch is the maximum number of messages to share at once.
 	maxShareBatch = 10
+	// peerCountLow is the target number of peers to connect to at any given time.
+	peerCountLow = 10
+	// peerCountHigh is the maximum number of peers to be connected to. If the
+	// number of connections exceeds this number, we will prune connections until
+	// we reach peerCountLow.
+	peerCountHigh = 12
```

Comment:
@fabioberger the Connection Manager decides which connections to prune based on tags/scores. This is good because it lets us easily implement something very similar to the "neighborhood establishment" algorithm described in the Architecture Doc.

Here's how it works: You can use [the `TagPeer` method](https://godoc.org/github.com/libp2p/go-libp2p-connmgr#BasicConnMgr.TagPeer) to add `val` to a peer's total score. The `tag` is simply a unique identifier (e.g. "0x-mesh-weijie-sub"). You can also use [the `UntagPeer` method](https://godoc.org/github.com/libp2p/go-libp2p-connmgr#BasicConnMgr.UntagPeer) to remove the `val` associated with a particular tag from a peer's total score. When it is time to prune connections, the Connection Manager takes the sum of the `val`s from all tags for each peer and disconnects the peers with the lowest score.

> Or will we be able to prune based on neighbor reputation? 

Yes, once we implement WeijieSub and peer ranking, all we need to do is add tags for each peer according to the "peer ranking" algorithm. As long as we keep everything in sync, it will automatically prune the connections with the lowest score.

To help find the right group of peers more quickly, I'll probably also add a slight positive score when a peer indicates that they speak the 0x Mesh protocol (we can detect this via a `StreamHandler`). This means the Connection Manager will automatically disconnect from peers that don't speak the protocol (or peers that only speak an older version). Thanks to @raulk for this suggestion on Slack.